### PR TITLE
Guide PL [skip ci]

### PR DESCRIPTION
**Pull request by bizley**
- REST error handling,
- structure entry scripts, 
- test overview,
- input validation,
- output client scripts

### CODE REVIEW

File: docs/guide-pl/input-validation.md [hunk 1]
```diff
@@ -0,0 +1,648 @@
...
+} else {
+    // akcja w przypadku niepoprawnej walidacji. Zmienna $errors jest tablicą zawierającą wiadomości błędów
+    $errors = $model->errors;
+}
+```
+
+
+## Deklaracja zasad <span id="declaring-rules"></span>
+
+Aby metoda `validate()` naprawdę zadziałała, powinieneś zadeklarować zasady walidacji dla atrybutów które mają jej podlegać.
```
Comment (SilverFire) [inline docs/guide-pl/input-validation.md]: `validate()` -> [[yii\base\Model::validate()|validate()]]

File: docs/guide-pl/input-validation.md [hunk 2]
```diff
@@ -0,0 +1,648 @@
...
+Powinno zostać to zrobione przez nadpisanie metody [[yii\base\Model::rules()]]. Poniższy przykład pokazuje jak zostały zadeklarowane zasady walidacji dla modelu `ContactForm`:
```
Comment (SilverFire) [inline docs/guide-pl/input-validation.md]: [[yii\base\Model::rules()]] -> [[yii\base\Model::rules()|rules()]]

File: docs/guide-pl/input-validation.md [hunk 3]
```diff
@@ -0,0 +1,648 @@
+Walidacja danych wejściowych
+================
+
+Jednak z głownych zasad mówi, że nigdy nie powinno ufać się danym otrzymanym od użytkowników oraz zawsze je walidować przez ich użyciem.
+
+Mamy [model](structure-models.md) wypełniony danymi od użytkownika. Możemy go zwalidować przez wywołanie metody [[yii\base\Model::validate()]].
```
Comment (SilverFire) [inline docs/guide-pl/input-validation.md]: [[yii\base\Model::validate()|validate()]]

File: docs/guide-pl/input-validation.md [hunk 4]
```diff
@@ -0,0 +1,648 @@
...
+
+* alias walidatora podstawowego, np. `required`, `in`, `date` itd. Zajrzyj do sekcji [Podstawowe walidatory](tutorial-core-validators.md) aby uzyskać pełną listę walidatorów podstawowych.
+* nazwa metody walidacji w klasie modelu, lub funkcja anonimowa. Po więcej szczegółów zajrzyj do sekcji [Inline Validators](#inline-validators).
+* pełna nazwa klasy walidatora. Po więcej szczegółów zajrzyj do sekcji [Standalone Validators](#standalone-validators).
+
+Zasada może zostać użyta do walidacji jednego lub wielu atrybutów, a atrybut może być walidowany przez jedną, lub wiele zasad.
+Zasada może zostać użyta dla konkretnych [scenariuszy](structure-models.md#scenarios) przez określenie opcji `on`.
+Jeśli nie określisz opcji `on` oznacza to, że zasada zostanie użyta w każdym scenariuszu.
+
+Kiedy zostaje wywołana metoda `validate()` zostają wykonane następujące kroki w celu wykonania walidacji:
```
Comment (SilverFire) [inline docs/guide-pl/input-validation.md]: [[yii\base\Model::validate()|validate()]]

File: docs/guide-pl/input-validation.md [hunk 5]
```diff
@@ -0,0 +1,648 @@
...
+
+1. Określenie które atrybuty powinny zostać zwalidowane przez pobranie listy atrybutów z [[yii\base\Model::scenarios()]] używając aktualnego [[yii\base\Model::scenario|scenario]].
```
Comment (SilverFire) [inline docs/guide-pl/input-validation.md]: [[yii\base\Model::scenarios()|scenarios()]]

File: docs/guide-pl/input-validation.md [hunk 6]
```diff
@@ -0,0 +1,648 @@
...
+    Wybrane atrybuty nazywane są *atrybutami aktywnymi*.
+2. Określenie które zasady walidacji powinny zostać użyte przez pobranie listy zasad z [[yii\base\Model::rules()]] używając aktualnego [[yii\base\Model::scenario|scenario]].
```
Comment (SilverFire) [inline docs/guide-pl/input-validation.md]: [[yii\base\Model::rules()|rules()]]

File: docs/guide-pl/input-validation.md [hunk 7]
```diff
@@ -0,0 +1,648 @@
...
+```
+
+Niektóre walidatory mogą wspierać dodatkowe wiadomości błedów aby bardziej precyzyjnie określić problemy przy walidacji.
+Dla przykładu, walidator [[yii\validators\NumberValidator|number]] wspiera [[yii\validators\NumberValidator::tooBig|tooBig]] oraz [[yii\validators\NumberValidator::tooSmall|tooSmall]] 
+do opisania kiedy liczba jest za duża lub za mała podczas walidacji. Możesz skonfigurować te wiadomości tak, jak pozostałe właściwości walidatorów w zasadzie walidacji.
+
+
+### Zdarzenia walidacji <span id="validation-events"></span>
+
+Podczas wywołania metody [[yii\base\Model::validate()]], zostaną wywołane dwie metody, które możesz nadpisać, aby dostosować proces walidacji:
```
Comment (SilverFire) [inline docs/guide-pl/input-validation.md]: [[yii\base\Model::validate()|validate()]]

**Comment by SilverFire** [comment]
Thank you very much for your effort.

I've posted notices for links to related methods. It's not, critical but if the context is about some class, we prefer to make link text shorter. Could you change do the same in your PR, please?

I don't know Polish, so I can't check spelling. @Yupik

**Comment by bizley** [comment]
I've made few corrections and shortened methods' links.
With next pull the rest of already translated PL doc will get all links shortened.

File: docs/guide-pl/input-validation.md [hunk 8]
```diff
@@ -0,0 +1,649 @@
...
+            $model->addError($attribute, $this->message);
+        }
+    }
+
+    public function clientValidateAttribute($model, $attribute, $view)
+    {
+        $statuses = json_encode(Status::find()->select('id')->asArray()->column());
+        $message = json_encode($this->message, JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
+        return <<<JS
+if ($.inArray(value, $statuses) > -1) {
```
Comment (SilverFire) [inline docs/guide-pl/input-validation.md]: Recently there was a PR that fixed this expression: https://github.com/yiisoft/yii2/commit/83d72cddc02da030a384e87295721ea6a2ce2937

**Comment by bizley** [comment]
Feel free to merge. No ETA on links shortening now.

**Comment by SilverFire** [comment]
Thank you very much for your work!

**Pull request merged**